### PR TITLE
[+] add quake api for subfinder

### PR DESCRIPTION
**Pull request by zer0yu**
Quake is a awesome cyberspace mapping engine, and you can use it to discover more subdomains. 
Like:
```
❯ ./subfinder -d skyscanner.net -sources quake

               __    _____           __         
   _______  __/ /_  / __(_)___  ____/ /__  _____
  / ___/ / / / __ \/ /_/ / __ \/ __  / _ \/ ___/
 (__  ) /_/ / /_/ / __/ / / / / /_/ /  __/ /    
/____/\__,_/_.___/_/ /_/_/ /_/\__,_/\___/_/ v2.4.8

                projectdiscovery.io

Use with caution. You are responsible for your actions
Developers assume no liability and are not responsible for any misuse or damage.
By using subfinder, you also agree to the terms of the APIs used.

[INF] Enumerating subdomains for skyscanner.net
analytics.skyscanner.net
ww2.business.skyscanner.net
carhirehelp.skyscanner.net
business.skyscanner.net
api.skyscanner.net
slipstream.skyscanner.net
ablink.sender.skyscanner.net
translate.skyscanner.net
hotelshelp.skyscanner.net
support.business.skyscanner.net
help.skyscanner.net
gateway.skyscanner.net
talent.skyscanner.net
cn.skyscanner.net
www.partners.skyscanner.net
forum.skyscanner.net
clicks.skyscanner.net
se.skyscanner.net
news.skyscanner.net
futuretech.skyscanner.net
partners.skyscanner.net
m.skyscanner.net
travelpro.skyscanner.net
yj.partners.api.skyscanner.net
azuremfa.skyscanner.net
widgets.skyscanner.net
www.skyscanner.net
redirect.skyscanner.net
partners.api.skyscanner.net
www.futuretech.skyscanner.net
[INF] Found 30 subdomains for skyscanner.net in 7 seconds 358 milliseconds
```
In config.yaml, you only need to add the following fields
```
resolvers:
    - 1.1.1.1
    - 1.0.0.1
    - 8.8.8.8
    - 8.8.4.4
    - 9.9.9.9
    - 9.9.9.10
    - 77.88.8.8
    - 77.88.8.1
    - 208.67.222.222
    - 208.67.220.220
sources:
    - quake
all-sources:
    - quake
quake: []
subfinder-version: 2.4.8
```

### CODE REVIEW

**Comment by ehsandeep** [comment]
@zer0yu thank you for adding the new source, I was looking to signup for keys from here - https://quake.360.cn/quake/login#/ correct me if I'm wrong, as per the form it's limited to signup with a local number?

**Comment by zer0yu** [comment]
> @zer0yu thank you for adding the new source, I was looking to signup for keys from here - https://quake.360.cn/quake/login#/ correct me if I'm wrong, as per the form it's limited to signup with a local number?

Yes, this is the correct url

**Comment by shadow1ng** [comment]
run error by using
./subfinder -d fofa.so -sources quake

``` 
Could not run source quake: quake.quakeResults.Data: []struct { Service struct { HTTP struct { Host string "json:\"host\""; StatusCode int "json:\"status_code\"" } "json:\"http\"" } }: Service: HTTP: StatusCode: readUint64: unexpected character: �, error found in #10 byte of ...|us_code":"\u6682\u65|..., bigger context ...|eaders":"\u6682\u65e0\u6743\u9650","status_code":"\u6682\u65e0\u6743\u9650","title":"\u9996\u9875","
```
image
image

**Comment by shadow1ng** [comment]
> run error by using
> ./subfinder -d fofa.so -sources quake
> 
> ```
> Could not run source quake: quake.quakeResults.Data: []struct { Service struct { HTTP struct { Host string "json:\"host\""; StatusCode int "json:\"status_code\"" } "json:\"http\"" } }: Service: HTTP: StatusCode: readUint64: unexpected character: �, error found in #10 byte of ...|us_code":"\u6682\u65|..., bigger context ...|eaders":"\u6682\u65e0\u6743\u9650","status_code":"\u6682\u65e0\u6743\u9650","title":"\u9996\u9875","
> ```
> 
> image
> image

解决办法
1.  StatusCode int `json:"status_code"`   修改成 StatusCode string    `json:"status_code"`  
2.  注释掉StatusCode int `json:"status_code"`

**Comment by zer0yu** [comment]
Thanks for the feedback, although I failed to replicate the bug on my MacOS Big Sur, I commented out the `StatusCode` field to ensure compatibility on most systems.

**Comment by ehsandeep** [comment]
@zer0yu not sure if it's too much to ask, is it possible for you to join https://discord.gg/projectdiscovery and ping "pdteam" or Twitter DM https://twitter.com/pdiscoveryio?  

This is related to obtaining a key to confirm this PR.

**Comment by zer0yu** [comment]
Thank you for the reminder @ehsandeep .

File: v2/pkg/subscraping/sources/quake/quake.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+)
+
+type quakeResults struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+	Data    []struct {
+		Service struct {
+			HTTP struct {
+				// Title           string `json:"title"`
```
Comment (Mzack9999) [inline v2/pkg/subscraping/sources/quake/quake.go:20]: I'd remove commented code if not used

**Review by Mzack9999 — APPROVED** [review]
I'd remove unused commented code - LGTM otherwise

File: v2/pkg/subscraping/sources/quake/quake.go (hunk 1 above)
Comment (zer0yu) [reply v2/pkg/subscraping/sources/quake/quake.go:20]: ok

**Comment by forgedhallpass** [comment]
Validated:

image

image

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**